### PR TITLE
solution exo345

### PR DESCRIPTION
**Pull request by tianyuchen**

### CODE REVIEW

File: Double_index/345-Reverse_Vowels_of_a_String.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        :rtype: str
+        """
+        vowels = ['a', 'e', 'i', 'o', 'u', 'A','E','I','O','U']
+
+        leftPointer = 0
+        rightPointer = len(s) - 1
+        s = list(s)
+
+        while leftPointer < rightPointer:
+            if s[leftPointer] not in vowels:
```
Comment (gameboy1024) [inline Double_index/345-Reverse_Vowels_of_a_String.py:34]: This lookup and the one below can be further optimized.

File: Double_index/345-Reverse_Vowels_of_a_String.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                leftPointer += 1
+            elif s[rightPointer] not in vowels:
+                rightPointer -= 1
+            else:
+                s[rightPointer], s[leftPointer] = s[leftPointer], s[rightPointer]
+                leftPointer += 1
+                rightPointer -= 1
+        return (''.join(s))
```
Comment (gameboy1024) [inline Double_index/345-Reverse_Vowels_of_a_String.py]: Surrounding parenthesis not needed.

**Review by gameboy1024 — CHANGES_REQUESTED** [review]

File: Double_index/345-Reverse_Vowels_of_a_String.py (hunk 2 above)
Comment (tianyuchen) [reply Double_index/345-Reverse_Vowels_of_a_String.py]: fixed :)

File: Double_index/345-Reverse_Vowels_of_a_String.py (hunk 1 above)
Comment (tianyuchen) [reply Double_index/345-Reverse_Vowels_of_a_String.py:34]: how can they be optimized?

Comment (gameboy1024) [reply Double_index/345-Reverse_Vowels_of_a_String.py:34]: https://stackoverflow.com/questions/513882/python-list-vs-dict-for-look-up-table

**Review by gameboy1024 — CHANGES_REQUESTED** [review]

File: Double_index/345-Reverse_Vowels_of_a_String.py (hunk 1 above)
Comment (tianyuchen) [reply Double_index/345-Reverse_Vowels_of_a_String.py:34]: I've converted list to set

**Review by gameboy1024 — APPROVED** [review]

**Pull request merged**